File: exams/frmSubjectsOutOf.vb
Public Class frmSubjectsOutOf

    Private Sub frmSubjectsOutOf_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        If Not connect() Then
            Me.Close()
        Else
            If stream = "All" Then
                query = "SELECT * FROM exam_results_subjects_out_of WHERE Examination='" & escape_string(exam_name) & "' AND term='" & tm & "' AND year='" & yr & "' AND Class='" & escape_string(class_form) & "'"
            Else
                query = "SELECT * FROM exam_results_subjects_out_of WHERE Examination='" & escape_string(exam_name) & "' AND term='" & tm & "' AND year='" & yr & "' AND Class='" & escape_string(class_form) & "' AND Stream='" & escape_string(stream) & "'"
            End If
            qread(query)
            Dim i As Integer = 0
            While dbreader.Read()
                For k As Integer = 0 To subjabb.Length - 1
                    dgvSubjects.Rows.Add()
                    dgvSubjects.Item("Abbreviation", k + (i * (subjabb.Length))).Value = subjabb(k)
                    dgvSubjects.Item("Subject", k + (i * (subjabb.Length))).Value = subjects(k)
                    dgvSubjects.Item("Clas", k + (i * (subjabb.Length))).Value = dbreader("Class")
                    dgvSubjects.Item("Str", k + (i * (subjabb.Length))).Value = dbreader("Stream")
                    dgvSubjects.Item("Marks", k + (i * (subjabb.Length))).Value = dbreader(subjname(k))
                Next
                i += 1
            End While
            If Not IsAdmin() Then
                For k As Integer = 0 To dgvSubjects.Rows.Count - 1
                    If IsSubjectTeacher(dgvSubjects.Item("Subject", k).Value, dgvSubjects.Item("Clas", k).Value, dgvSubjects.Item("Str", k).Value, tm, yr) Then
                        dgvSubjects.Rows(k).ReadOnly = True
                    Else
                        dgvSubjects.Rows(k).ReadOnly = False
                    End If
                Next
            End If
        End If
    End Sub

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        start()
        For k As Integer = 0 To dgvSubjects.Rows.Count - 1
            If Not qwrite("UPDATE exam_results_subjects_out_of SET `" & dgvSubjects.Item("Abbreviation", k).Value & "`='" & dgvSubjects.Item("Marks", k).Value & "' WHERE (Examination='" & escape_string(exam_name) & "' AND term='" & tm & "' AND year='" & yr & "' AND Class='" & escape_string(dgvSubjects.Item("Clas", k).Value) & "' AND Stream='" & escape_string(dgvSubjects.Item("Str", k).Value) & "')") Then
                rollback()
                failure("Subject Total Marks Could Not Be Saved!")
                Exit Sub
            End If
        Next
        commit()
        success("Subject Total Marks Successfully Saved!")
    End Sub
End Class

File: exams/frmSubjectsOutOf.cs
using System;
using Microsoft.VisualBasic.CompilerServices;

namespace exams
{
    public partial class frmSubjectsOutOf
    {
        public frmSubjectsOutOf()
        {
            InitializeComponent();
            _Button1.Name = "Button1";
        }

        private void frmSubjectsOutOf_Load(object sender, EventArgs e)
        {
            if (!publicSubsNFunctions.connect())
            {
                Close();
            }
            else
            {
                if (publicSubsNFunctions.stream == "All")
                {
                    publicSubsNFunctions.query = "SELECT * FROM exam_results_subjects_out_of WHERE Examination='" + publicSubsNFunctions.escape_string(publicSubsNFunctions.exam_name) + "' AND term='" + publicSubsNFunctions.tm + "' AND year='" + publicSubsNFunctions.yr + "' AND Class='" + publicSubsNFunctions.escape_string(publicSubsNFunctions.class_form) + "'";
                }
                else
                {
                    publicSubsNFunctions.query = "SELECT * FROM exam_results_subjects_out_of WHERE Examination='" + publicSubsNFunctions.escape_string(publicSubsNFunctions.exam_name) + "' AND term='" + publicSubsNFunctions.tm + "' AND year='" + publicSubsNFunctions.yr + "' AND Class='" + publicSubsNFunctions.escape_string(publicSubsNFunctions.class_form) + "' AND Stream='" + publicSubsNFunctions.escape_string(publicSubsNFunctions.stream) + "'";
                }

                publicSubsNFunctions.qread(ref publicSubsNFunctions.query);
                int i = 0;
                while (publicSubsNFunctions.dbreader.Read())
                {
                    for (int k = 0, loopTo = publicSubsNFunctions.subjabb.Length - 1; k <= loopTo; k++)
                    {
                        dgvSubjects.Rows.Add();
                        dgvSubjects["Abbreviation", k + i * publicSubsNFunctions.subjabb.Length].Value = publicSubsNFunctions.subjabb[k];
                        dgvSubjects["Subject", k + i * publicSubsNFunctions.subjabb.Length].Value = publicSubsNFunctions.subjects[k];
                        dgvSubjects["Clas", k + i * publicSubsNFunctions.subjabb.Length].Value = publicSubsNFunctions.dbreader["Class"];
                        dgvSubjects["Str", k + i * publicSubsNFunctions.subjabb.Length].Value = publicSubsNFunctions.dbreader["Stream"];
                        dgvSubjects["Marks", k + i * publicSubsNFunctions.subjabb.Length].Value = publicSubsNFunctions.dbreader(publicSubsNFunctions.subjname[k]);
                    }

                    i += 1;
                }

                if (Conversions.ToBoolean(!publicSubsNFunctions.IsAdmin()))
                {
                    for (int k = 0, loopTo1 = dgvSubjects.Rows.Count - 1; k <= loopTo1; k++)
                    {
                        if (publicSubsNFunctions.IsSubjectTeacher(Conversions.ToString(dgvSubjects["Subject", k].Value), Conversions.ToString(dgvSubjects["Clas", k].Value), Conversions.ToString(dgvSubjects["Str", k].Value), publicSubsNFunctions.tm, publicSubsNFunctions.yr.ToString()))
                        {
                            dgvSubjects.Rows[k].ReadOnly = true;
                        }
                        else
                        {
                            dgvSubjects.Rows[k].ReadOnly = false;
                        }
                    }
                }
            }
        }

        private void Button1_Click(object sender, EventArgs e)
        {
            publicSubsNFunctions.start();
            for (int k = 0, loopTo = dgvSubjects.Rows.Count - 1; k <= loopTo; k++)
            {
                if (!publicSubsNFunctions.qwrite(Conversions.ToString(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject("UPDATE exam_results_subjects_out_of SET `", dgvSubjects["Abbreviation", k].Value), "`='"), dgvSubjects["Marks", k].Value), "' WHERE (Examination='"), publicSubsNFunctions.escape_string(publicSubsNFunctions.exam_name)), "' AND term='"), publicSubsNFunctions.tm), "' AND year='"), publicSubsNFunctions.yr), "' AND Class='"), publicSubsNFunctions.escape_string(Conversions.ToString(dgvSubjects["Clas", k].Value))), "' AND Stream='"), publicSubsNFunctions.escape_string(Conversions.ToString(dgvSubjects["Str", k].Value))), "')"))))
                {
                    publicSubsNFunctions.rollback();
                    publicSubsNFunctions.failure("Subject Total Marks Could Not Be Saved!");
                    return;
                }
            }

            publicSubsNFunctions.commit();
            publicSubsNFunctions.success("Subject Total Marks Successfully Saved!");
        }
    }
}